src/coreclr/inc/corinfo.h

```diff
@@ -1616,10 +1616,12 @@ struct CORINFO_DEVIRTUALIZATION_INFO
     //      invariant is `resolveVirtualMethod(...) == (devirtualizedMethod != nullptr)`.
     // - requiresInstMethodTableArg is set to TRUE if the devirtualized method requires a type handle arg.
     // - exactContext is set to wrapped CORINFO_CLASS_HANDLE of devirt'ed method table.
+    // - failure reason set if devirtualization fails.
     //
     CORINFO_METHOD_HANDLE       devirtualizedMethod;
     bool                        requiresInstMethodTableArg;
     CORINFO_CONTEXT_HANDLE      exactContext;
+    const char*                 failureReason;
```

Comment:
Please describe the memory behavior of this pointer. In particular, is it expected to be remain across multiple calls to resolveVirtulaMethod?

Also, superpmi is not currently storing/restoring this value.

Reply:
Currently it is only used immediately after the call.

I suppose it would be simpler to make it an enum. I made it a string because I anticipated there would be a lot of churn as we evolve this and didn't want to also have to rev the jit interface each time. 

I don't expect actual behavior of the jit to depend on this -- it is just there for diagnostics -- so perhaps an enum with an understanding that a given jit build might see values outside its known enum range will work.

Reply:
A string is a fine choice. I just want to document its memory behavior, and used directly after the call is a simple and fairly common behavior in the jit interface. The only concern is that handling of strings across the managed/native boundary needs to have some care taken around object lifetime, etc, and writing down the contract is necessary.

Reply:
I went ahead and changed it over to an enum, it's simple for SPMI to deal with.